### PR TITLE
Move Last sync info into database instead of local filesystem

### PR DESCRIPTION
**Pull request by bergmania**
TBD

### CODE REVIEW

File: src/Umbraco.Infrastructure/Sync/LastSyncedDatabaseManager.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        {
+            // We sitename changes doing runtime. we save the value to the new name.
+            var lastSyncedId = _lastSyncedId ?? _lastSyncedDatabaseRepository.GetValue(ServerKey);
+            _hostingSettings = hostingSettings;
+
+            _lastSyncedDatabaseRepository.SetValue(ServerKey, lastSyncedId);
+        });
+    }
+
+    private string ServerKey
```
Comment (ronaldbarendse) [inline src/Umbraco.Infrastructure/Sync/LastSyncedDatabaseManager.cs:34]: Isn't this the same key as used in the `umbracoServer` table and can't this be stored in a new column, removing the need for adding a new table?

Comment (bergmania) [reply src/Umbraco.Infrastructure/Sync/LastSyncedDatabaseManager.cs:34]: Machine name is part of the key, which is a part of the problem, when Azure shift worker you get a new machine name.

Comment (ronaldbarendse) [reply src/Umbraco.Infrastructure/Sync/LastSyncedDatabaseManager.cs:34]: Ah right, so isn't this a similar problem that resulted in adding a `MainDomKeyDiscriminator` in PR https://github.com/umbraco/Umbraco-CMS/pull/12037? Although for the `MainDomId` the machine name and application ID are added (which is what I believe is used in the `umbracoServer` table), but if this discriminator value would serve the same purpose, we might want to use this value instead and avoid requiring users to set multiple values to get things working correctly?

Comment (nzdev) [reply src/Umbraco.Infrastructure/Sync/LastSyncedDatabaseManager.cs:34]: Alternatively, why not store the sync state in nucache itself. Then it doesn't matter if files on disc get out of sync with it's state in the sql database. It's already transactional, so the last step of a nucache write could write the lastsyncid.

Comment (nzdev) [reply src/Umbraco.Infrastructure/Sync/LastSyncedDatabaseManager.cs:34]: It's used for multiple things, though ideally it would be split.

**Pull request closed without merging**